tests/compile-fail/ptr_int_guess.rs

```diff
@@ -0,0 +1,16 @@
+// compile-flags: -Zmiri-disable-stacked-borrows -Zmiri-seed=0
+
+fn ptr_int_guess() {
+    let x: i32 = 3;
+    // we fix the seed so that this pointer's value is 0x23080
```

Comment:
The value can change each time we update the standard library, so this test seems too fragile.

Without strict provenance and number validity checking, you can transmute a pointer to an integer and add 0 to get a "proper" integer without exposing the pointer. Then you can use that to implement a similar test.

`ptr_int_unexposed` might be a clearer name.

Reply:
Oh and can you add a test that round-trips a function pointer through an integer?